src/flash-source-buffer.js

```diff
@@ -260,58 +252,35 @@ export default class FlashSourceBuffer extends videojs.EventTarget {
       return;
     }
 
-    do {
-      appendIterations++;
-      // concatenate appends up to the max append size
-      chunk = this.buffer_[0].subarray(startByte, startByte + this.chunkSize_);
+    // concatenate appends up to the max append size
+    let chunk = this.buffer_[0].subarray(0, chunkSize);
 
-      // requeue any bytes that won't make it this round
-      if (chunk.byteLength < this.chunkSize_ ||
-          this.buffer_[0].byteLength === startByte + this.chunkSize_) {
-        startByte = 0;
-        this.buffer_.shift();
-      } else {
-        startByte += this.chunkSize_;
-      }
+    // requeue any bytes that won't make it this round
+    if (chunk.byteLength < chunkSize ||
+        this.buffer_[0].byteLength === chunkSize) {
+      this.buffer_.shift();
+    } else {
+      this.buffer_[0] = this.buffer_[0].subarray(chunkSize);
+    }
 
-      this.bufferSize_ -= chunk.byteLength;
+    this.bufferSize_ -= chunk.byteLength;
 
-      // base64 encode the bytes
-      binary = '';
-      length = chunk.byteLength;
-      for (i = 0; i < length; i++) {
-        binary += String.fromCharCode(chunk[i]);
-      }
-      b64str = window.btoa(binary);
-
-      // bypass normal ExternalInterface calls and pass xml directly
-      // IE can be slow by default
-      this.mediaSource_.swfObj.CallFunction(
-        '<invoke name="vjs_appendBuffer"' +
-        'returntype="javascript"><arguments><string>' +
-        b64str +
-        '</string></arguments></invoke>'
-      );
-      appendTime = (new Date()) - startTime;
-    } while (this.buffer_.length &&
-        appendTime < FlashConstants.TIME_PER_TICK);
-
-    if (this.buffer_.length && startByte) {
-      this.buffer_[0] = this.buffer_[0].subarray(startByte);
-    }
+    // base64 encode the bytes
+    let binary = '';
+    let length = chunk.byteLength;
 
-    if (appendTime >= FlashConstants.TIME_PER_TICK) {
-      // We want to target 4 iterations per time-slot so that gives us
-      // room to adjust to changes in Flash load and other externalities
-      // such as garbage collection while still maximizing throughput
-      this.chunkSize_ = Math.floor(this.chunkSize_ * (appendIterations / 4));
+    for (let i = 0; i < length; i++) {
+      binary += String.fromCharCode(chunk[i]);
     }
-
-    // We also make sure that the chunk-size doesn't drop below 1KB or
-    // go above 1MB as a sanity check
-    this.chunkSize_ = Math.max(
-      FlashConstants.MIN_CHUNK,
-      Math.min(this.chunkSize_, FlashConstants.MAX_CHUNK));
+    let b64str = window.btoa(binary);
+
+    // bypass normal ExternalInterface calls and pass xml directly
```

Comment:
Yeah... it was back in the day. I don't know if the performance of ExternalInterface calls have improved in recent times.